### PR TITLE
Sledgehammer updates

### PR DESCRIPTION
**Pull request by sygibson**
Updates sledgehammer to add `/etc/issue` and `/etc/issue.net` appropriate signature output.

- kickstarts `%post` section creates a fake sha1sum
- writes `/etc/sledgehammer-release` and `/etc/sledgehammer-id` files
- updates to the appropriate bits to use newly generated ID

Tested to work correctly.

### CODE REVIEW

File: sledgehammer-builder/tasks/sledgehammer-stage-bits.yaml [hunk 1]
```diff
@@ -590,9 +604,11 @@ Templates:
     Contents: |
       #!/bin/bash
       {{ template "setup.tmpl" .}}
-      echo "Generating unique signature for Sledgehammer"
+      #echo "Generating unique signature for Sledgehammer"
```
Comment (galthaus) [inline sledgehammer-builder/tasks/sledgehammer-stage-bits.yaml]: Just delete this

File: sledgehammer-builder/tasks/sledgehammer-stage-bits.yaml [hunk 2]
```diff
@@ -590,9 +604,11 @@ Templates:
...
       cd /IMAGE
-      signature="$(sha1sum <root.squashfs |awk '{print $1}')"
+      #signature="$(sha1sum <root.squashfs |awk '{print $1}')"
```
Comment (galthaus) [inline sledgehammer-builder/tasks/sledgehammer-stage-bits.yaml]: Just delete this.

File: content/bootenvs/centos-7.6.1810.yml [hunk 3]
```diff
@@ -17,13 +25,14 @@ OS:
         ks={{.Machine.Url}}/compute.ks
         method={{.Env.InstallUrl}}
         inst.geoloc=0
+        {{if .ParamExists "kernel-options"}}{{.Param "kernel-options"}}{{end}}
         --
         {{if .ParamExists "kernel-console"}}{{.Param "kernel-console"}}{{end}}
     aarch64:
       Loader: "grubarm64.efi"
-      IsoFile: CentOS-7-aarch64-Minimal-1810.iso
+      IsoFile: "CentOS-7-aarch64-Minimal-1810.iso"
```
Comment (VictorLowther) [inline content/bootenvs/centos-7.6.1810.yml]: Why are quotes being added here?  YAML does not need them to infer that this is a string.

File: content/bootenvs/ubuntu-18.04-arm64-hwe.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  Family: "ubuntu"
+  Codename: "Bionic Beaver"
+  Version: "18.04"
+  SupportedArchitectures:
+    arm64:
+      IsoFile: "ubuntu-18.04.2-server-arm64.iso"
+      IsoUrl: "http://cdimage.ubuntu.com/releases/18.04/release/ubuntu-18.04.2-server-arm64.iso"
+      IsoSha256: "22ffc947df8c699920582c6d446b4e32e8471c1684bdeab7c91e36c1ebb7f2ba"
+      Kernel: "install/hwe-netboot/ubuntu-installer/arm64/linux"
+      Loader: "bootnetaa64.efi"
```
Comment (VictorLowther) [inline content/bootenvs/ubuntu-18.04-arm64-hwe.yml]: I really think you want grubarm64.efi here

File: content/bootenvs/ubuntu-18.04.yml [hunk 5]
```diff
@@ -1,30 +1,69 @@
...
+        rw quiet
+        {{if .ParamExists "kernel-options"}}{{.Param "kernel-options"}}{{end}}
+        --
+        {{if .ParamExists "kernel-console"}}{{.Param "kernel-console"}}{{end}}
+    arm64:
+      IsoFile: "ubuntu-18.04.2-server-arm64.iso"
+      IsoUrl: "http://cdimage.ubuntu.com/releases/18.04/release/ubuntu-18.04.2-server-arm64.iso"
+      IsoSha256: "22ffc947df8c699920582c6d446b4e32e8471c1684bdeab7c91e36c1ebb7f2ba"
+      Kernel: "install/netboot/ubuntu-installer/arm64/linux"
+      Loader: "bootnetaa64.efi"
```
Comment (VictorLowther) [inline content/bootenvs/ubuntu-18.04.yml]: You also probably want grubarm64.efi here too.

File: sledgehammer-builder/bootenvs/build-sledgehammer.yaml [hunk 6]
```diff
@@ -170,6 +170,12 @@ Templates:
       %end
 
       %post
+      N=10
+      ID=$(while (( N > 0 )); do uuidgen; (( N-- )); done | sha1sum | awk ' { print $1 } ')
```
Comment (VictorLowther) [inline sledgehammer-builder/bootenvs/build-sledgehammer.yaml]: I get what you are doing here, but that is a weird way of shoving random numbers through sha1sum.

File: content/bootenvs/centos-7.6.1810.yml (hunk 3 above)
Comment (sygibson) [reply content/bootenvs/centos-7.6.1810.yml]: **_quotes:_** we have wild inconsistencies about quoting or not quoting YAML.  I prefer all things quoted except things that are required to be bare - if some strings are quoted, all strings should be quoted for consistency

File: content/bootenvs/ubuntu-18.04-arm64-hwe.yml (hunk 4 above)
Comment (sygibson) [reply content/bootenvs/ubuntu-18.04-arm64-hwe.yml]: **_quoting:_**  I prefer all things quoted unless they are required to be bare - in either case, I hate inconsistency in (at least) a given content pack ... 

**_boot loader:_** In Ubuntu 18.04 there is no `grubarm64.efi`

wget http://cdimage.ubuntu.com/releases/18.04/release/ubuntu-18.04.2-server-arm64.iso
mkdir mnt
mount -o loop ubuntu-18.04.2-server-arm64.iso mnt
cd mnt
find . -type f | grep "\.efi$"
# outputs:
./install/hwe-netboot/ubuntu-installer/arm64/bootnetaa64.efi
./install/netboot/ubuntu-installer/arm64/bootnetaa64.efi

File: content/bootenvs/ubuntu-18.04.yml (hunk 5 above)
Comment (sygibson) [reply content/bootenvs/ubuntu-18.04.yml]: comment on boot loader for `content/bootenvs/ubuntu-18.04-arm64-hwe.yml`

File: content/bootenvs/ubuntu-18.04-arm64-hwe.yml (hunk 4 above)
Comment (galthaus) [reply content/bootenvs/ubuntu-18.04-arm64-hwe.yml]: Then don't we need arm64/bootnetaa64.efi or the fuller path?

Comment (galthaus) [reply content/bootenvs/ubuntu-18.04-arm64-hwe.yml]: Okay - looked it up.  Loader is used by the DHCP server to override bootfile for the bootenv.  It is relative to the top of the tftproot directory and grubarm64.efi is provided by DRP by default.

**Pull request merged**